src/main/java/io/nats/client/impl/JsPrefixManager.java

```diff
@@ -0,0 +1,49 @@
+package io.nats.client.impl;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static io.nats.client.impl.NatsJetStreamConstants.JSAPI_PREFIX;
+import static io.nats.client.impl.NatsJetStreamConstants.JS_PREFIX;
+import static io.nats.client.support.NatsConstants.DOT;
+import static io.nats.client.support.Validator.nullOrEmpty;
+import static io.nats.client.support.Validator.validateJetStreamPrefix;
+
+public class JsPrefixManager {
+
+    private static final Set<String> JS_PREFIXES = Collections.synchronizedSet(new HashSet<>());
+
+    public static String addPrefix(String prefix) {
+        if (nullOrEmpty(prefix)) {
+            return JSAPI_PREFIX;
+        }
+
+        prefix = validateJetStreamPrefix(prefix);
+        if (!prefix.endsWith(DOT)) {
+            prefix += DOT;
+        }
+
+        JS_PREFIXES.add(prefix);
+
+        return prefix;
+    }
+
+    public static boolean isJsMessage(final String replyTo) {
+        if (replyTo != null) {
+            if (replyTo.startsWith(JS_PREFIX)) {
+                return true;
+            }
+
+            synchronized (JS_PREFIXES) {
```

Comment:
Wish there was a way to avoid this... this is in the fastpath, but I guess need to do this when there are multiple prefixes.  

Could use a synchronized list, get the count (synchronized) and then iterate through up to the count since prefixes will never change order or be deleted once created.

Can optimize later, if need be....

Reply:
I am using a synchronized list. In the javadoc itself, it says if you iterate the list, you must synchronize it. I suppose I could make a read only copy after every update.

Reply:
for js messages from the server, the majority I think, this is going to be $JS.ACK, so won't even go into the loop checking.

Reply:
Good idea, implementing

Reply:
let's leave this for now. I think it might be more expensive to know the count and get(index) b/c each get index will do a synchronized so this will just have one for the entire list instead of each item.